### PR TITLE
Add additional Orbit methods and properties from KSP's Orbit API

### PR DESCRIPTION
**Pull request by jwvanderbeck**

### CODE REVIEW

**Comment by jwvanderbeck** [comment]
Reference #303

File: service/SpaceCenter/src/Services/Orbit.cs [hunk 1]
```diff
@@ -286,5 +286,75 @@ public static Tuple3 ReferencePlaneDirection (ReferenceFrame referenceFrame)
...
+            return InternalOrbit.EccentricAnomalyAtUT (UT);
+        }
+
+        /// <summary>
+        /// Gets the orbital speed at time.
+        /// </summary>
+        /// <returns>The orbital speed at the given time.</returns>
+        /// <param name="time">Time from now.</param>
+        [KRPCMethod]
+        public double getOrbitalSpeedAt (double time) {
```
Comment (djungelorm) [inline service/SpaceCenter/src/Services/Orbit.cs]: Rename this to ```OrbitalSpeedAt``` to be consistent

File: service/SpaceCenter/src/Services/Orbit.cs [hunk 2]
```diff
@@ -286,5 +286,75 @@ public static Tuple3 ReferencePlaneDirection (ReferenceFrame referenceFrame)
...
+        }
+
+        /// <summary>
+        /// Gets the UT for given true anomaly.
+        /// </summary>
+        /// <returns>The UT for true anomaly.</returns>
+        /// <param name="tA">True Anomaly.</param>
+        /// <param name="wrapAfterSeconds">Seconds after which to wrap the orbit</param>
+        [KRPCMethod]
+        public double GetUTforTrueAnomaly (double tA, double wrapAfterSeconds) {
```
Comment (djungelorm) [inline service/SpaceCenter/src/Services/Orbit.cs]: Rename this to ```UTForTrueAnomaly``` to be consistent?

File: service/SpaceCenter/src/Services/Orbit.cs [hunk 3]
```diff
@@ -286,5 +286,75 @@ public static Tuple3 ReferencePlaneDirection (ReferenceFrame referenceFrame)
...
+        public double getOrbitalSpeedAt (double time) {
+            return InternalOrbit.getOrbitalSpeedAt (time);
+        }
+
+        /// <summary>
+        /// Gets the current orbital speed.
+        /// </summary>
+        /// <value>The orbital speed.</value>
+        [KRPCProperty]
+        public double orbitalSpeed {
```
Comment (djungelorm) [inline service/SpaceCenter/src/Services/Orbit.cs]: Should be camel cased to "OrbitalSpeed"

File: service/SpaceCenter/src/Services/Orbit.cs [hunk 4]
```diff
@@ -286,5 +286,75 @@ public static Tuple3 ReferencePlaneDirection (ReferenceFrame referenceFrame)
...
+        public double TrueAnomalyAtUT (double UT) {
+            return InternalOrbit.TrueAnomalyAtUT (UT);
+        }
+
+        /// <summary>
+        /// Gets the UT for given true anomaly.
+        /// </summary>
+        /// <returns>The UT for true anomaly.</returns>
+        /// <param name="tA">True Anomaly.</param>
+        /// <param name="wrapAfterSeconds">Seconds after which to wrap the orbit</param>
```
Comment (djungelorm) [inline service/SpaceCenter/src/Services/Orbit.cs]: What does this mean?

**Comment by djungelorm** [comment]
Thanks for the PR! Just some minor inline comments. Also ignore the travis failures, it's due to the documentation but I can patch that up when I merge the changes.

File: service/SpaceCenter/src/Services/Orbit.cs (hunk 2 above)
Comment (jwvanderbeck) [inline service/SpaceCenter/src/Services/Orbit.cs]: Consistent with what?

File: service/SpaceCenter/src/Services/Orbit.cs (hunk 4 above)
Comment (jwvanderbeck) [inline service/SpaceCenter/src/Services/Orbit.cs]: I have no clue :(

**Comment by jwvanderbeck** [comment]
Out of curiosity what about the documentation is causing the failure?

File: service/SpaceCenter/src/Services/Orbit.cs (hunk 2 above)
Comment (djungelorm) [inline service/SpaceCenter/src/Services/Orbit.cs]: The other methods, as they are not prefixed with "Get"

**Comment by djungelorm** [comment]
To get the build working you need to add the new RPCs to file ```doc/order.txt```. The travis build output suggests these are missing:
```
SpaceCenter.Orbit.TrueAnomalyAtRadius
SpaceCenter.Orbit.OrbitalSpeedAtTime
SpaceCenter.Orbit.TrueAnomalyAtUT
SpaceCenter.Orbit.RadiusAtTrueAnomaly
SpaceCenter.Orbit.EccentricAnomalyAtUT
SpaceCenter.Orbit.UTAtTrueAnomaly
SpaceCenter.Orbit.OrbitalSpeed
```

You'll also need to remove the ```<value>...</value>``` tag from the documentation for OrbitalSpeed as it's not understood by the documentation generator (and IMO a bit unnecessary as it says the same thing as the summary!)

**Pull request merged**